test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,306 @@
+package docker_e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"celestiaorg/celestia-app/test/docker-e2e/dockerchain"
+
+	"github.com/celestiaorg/tastora/framework/testutil/wait"
+	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+)
+
+// TallyResponse represents the JSON response from the signal tally query
+// We had to define it again because the JSON response type is not a number but string
+type TallyResponse struct {
+	VotingPower      string `json:"voting_power"`
+	ThresholdPower   string `json:"threshold_power"`
+	TotalVotingPower string `json:"total_voting_power"`
+}
+
+// TestCelestiaAppMinorUpgrade tests a simple upgrade from one minor version to another by swapping the binary.
+func (s *CelestiaTestSuite) TestCelestiaAppMinorUpgrade() {
+	if testing.Short() {
+		s.T().Skip("skipping celestia-app minor upgrade test in short mode")
+	}
+
+	tt := []struct {
+		Name                string
+		BaseBinaryVersion   string
+		TargetBinaryVersion string
+	}{
+		{
+			Name:                "v4.0.2-rc2 to v4.0.10",
+			BaseBinaryVersion:   "v4.0.2-rc2",
+			TargetBinaryVersion: "v4.0.10",
+		},
+		{
+			Name:                "v4.0.9-mocha to v4.0.10-mocha",
+			BaseBinaryVersion:   "v4.0.9-mocha",
+			TargetBinaryVersion: "v4.0.10-mocha",
+		},
+		{
+			Name:                "v4.0.9-arabica to v4.0.10-arabica",
+			BaseBinaryVersion:   "v4.0.9-arabica",
+			TargetBinaryVersion: "v4.0.10-arabica",
+		},
```

Comment:
[blocking question]

does this test use the current branch (the PR) and run alongside the older versions? its very unclear to me that that is occurring if it is. 

I have a very limited understanding of what is occurring under the hood here, so I could definitely be missing something. It looks like we're just running the exact same test each PR and not using the branch that we're curious in testing? if that's true, why do that? it seems to defeat the purpose of the test.